src/iop/lut3d.c

```diff
@@ -378,33 +378,45 @@ uint8_t calculate_clut_haldclut(char *filepath, float **clut)
     return 0;
   }
   level *= level;  // to be equivalent to cube level
+  if(level > 256)
+  {
+    fprintf(stderr, "[lut3d] error - LUT 3D size %d > 256\n", level);
+    dt_control_log(_("error - lut 3D size %d exceeds the maximum supported"), level);
+    fclose(png.f);
+    png_destroy_read_struct(&png.png_ptr, &png.info_ptr, NULL);
+    return 0;
+  }
   const size_t buf_size = (size_t)png.height * png_get_rowbytes(png.png_ptr, png.info_ptr);
   dt_print(DT_DEBUG_DEV, "[lut3d] allocating %zu bytes for png file\n", buf_size);
   uint8_t *buf = NULL;
   buf = dt_alloc_align(16, buf_size);
   if(!buf)
   {
-    fclose(png.f);
-    png_destroy_read_struct(&png.png_ptr, &png.info_ptr, NULL);
     fprintf(stderr, "[lut3d] error - allocating buffer for png lut\n");
     dt_control_log(_("error - allocating buffer for png lut"));
+    fclose(png.f);
+    png_destroy_read_struct(&png.png_ptr, &png.info_ptr, NULL);
     return 0;
   }
   if (read_image(&png, buf))
   {
-    dt_free_align(buf);
     fprintf(stderr, "[lut3d] error - could not read png image `%s'\n", filepath);
     dt_control_log(_("error - could not read png image %s"), filepath);
+    dt_free_align(buf);
+    fclose(png.f);
```

Comment:
After read image this is not needed and indeed crash dt.

Reply:
Sorry, I remove that. Doesn't crash on my system ...